### DISCUSSION TITLE
Maze of different shapes

### DISCUSSION THREAD

**Discussion by botcheddevil** [john-science/mazelib#133] [Q&A]
Right now we have all the mazes here in square shape only, could these shapes be in circle, hexagon or rectangle?

**Comment by john-science** [comment]
An interesting point!  Yeah, all my default mazes start out as rectangles. 

I think this will involve writing a little code. Essentially, you'd want to be able to pass in a (still-rectangular) grid, like we do in [DungeonRooms](https://github.com/john-science/mazelib/blob/main/mazelib/generate/DungeonRooms.py), to a maze-generating algorithm.  

But inside the maze rectangle, you just fill in the `Maze.grid` with a bunch of `1`s, so you end up with the perimeter shape you want.

It sounds easy enough. Do you want to take a crack at it, or do you want me to give it a whirl?

**Comment by botcheddevil** [comment]
Not sure if you have looked into at this https://www.mazegenerator.net

Its just not a perimeter shape, the internal nodes and edges also have conform to the to it.

**Reply by john-science** [reply]
That's interesting!

At the moment, `mazelib` only supports rectangular grids. 

What I wouldn't want to do is have to make variants for every generator, solver, and transmuter, to support different shapes. That wouldn't be supportable.

**Comment by botcheddevil** [comment]
May be we can arrange the  nodes such, so that each node has 6 exit / entry points where some edges are locked / walled and others are not in case of hexagon, not sure how we would implement a circle.

I haven't looked deep into this repo, I understand here we get a matrix of 1's and 0's  which represent wall and path respectively.
